cmd/id-exporter/main.go

```diff
@@ -24,15 +24,56 @@ type idExporter struct {
 	grace time.Duration
 }
 
-type id struct {
+// resultEntry is a JSON marshalable exporter result entry.
+type resultEntry struct {
+	// ID is exported to support marshaling to JSON.
 	ID int64 `json:"id"`
+
+	// Hostname is exported to support marshaling to JSON. Not all queries
+	// will fill this field, so it's JSON field tag marks at as
+	// omittable.
+	Hostname string `json:"hostname,omitempty"`
+}
+
+// reverseHostname converts (reversed) names sourced from the
+// registrations table to standard hostnames.
+func (r *resultEntry) reverseHostname() {
+	r.Hostname = sa.ReverseName(r.Hostname)
+}
+
+// idExporterResults is a container for a slice of pointers to
+// `resultEntry` objects. The `ids` field is passed as a selectable
+// 'holder' in calls to `dbMap.Select()`.
+type idExporterResults struct {
+	ids []*resultEntry
+}
+
+// makeJSON returns JSON as bytes for all elements of the inner `id`
+// slice.
+func (i *idExporterResults) makeJSON() ([]byte, error) {
```

Comment:
any reason this method isn't named `Marshal`?

Reply:
😅 that's a good idea 👍🏻